magento/pwa-studio
# Lazy Load AuthModal

## Description

The `AuthModal` component can be lazy loaded, as guest users may never use it.

This saves us ~6.5 KB on initial load of the application.

## Related Issue





Closes PWA-113.

## Acceptance 



@zetlen 
### Verification Stakeholders

### Specification



### Verification Steps


1. `yarn watch:venia`
1. Open Venia in the browser
1. Open the left (navigation) drawer
1. View Browser's "Network" tab
1. Click the "Sign In" button
1. Verify that new entries appeared in the Network tab (downloading `AuthModal` chunk)
1. Verify that the `AuthModal` chunk loads and the app works as expected

You can also verify that the loading spinner appears by artificially slowing the network speed in the Network tab before clicking the Sign In button.

## Screenshots / Screen Captures (if appropriate)

![Screen Shot 2019-12-04 at 10 42 13 AM](https://user-images.githubusercontent.com/13182778/70162313-0b0ab780-1683-11ea-8c5e-5ee1310c6f6b.png)

![Screen Shot 2019-12-04 at 10 32 17 AM](https://user-images.githubusercontent.com/13182778/70161307-7b183e00-1681-11ea-9ea8-5707e98337b4.png)
![Screen Shot 2019-12-04 at 10 33 00 AM](https://user-images.githubusercontent.com/13182778/70161308-7bb0d480-1681-11ea-90c1-fb89161e288b.png)

## Checklist

* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.
